src/interface/variable.hpp

```diff
@@ -204,6 +204,40 @@ class EdgeVariable {
   std::string label_;
 };
 
+template <typename T>
+class ParticleVariable {
+ public:
+  /// Initialize a particle variable
+  ParticleVariable(const std::string label, const int npool, const Metadata &metadata)
+      : data(label, npool), npool_(npool), m_(metadata), label_(label) {}
+
+  // accessors
+  KOKKOS_FORCEINLINE_FUNCTION
+  ParArrayND<T> &Get() { return data; }
+  template <typename... Args>
+  KOKKOS_FORCEINLINE_FUNCTION T &operator()(Args... args) { // const {
+    return data(std::forward<Args>(args)...);
+  }
+
+  ///< retrieve metadata for variable
+  const Metadata metadata() const { return m_; }
+
+  bool IsSet(const MetadataFlag bit) const { return m_.IsSet(bit); }
+
+  ///< retrieve label for variable
+  std::string label() { return label_; }
```

Comment:
Make const noexcept

Reply:
I'll make this `const` but I'm not comfortable using `noexcept` so especially for still-under-development code that isn't performance critical here and elsewhere I'd rather leave that off in this PR. Other `label()` functions in this file aren't `noexcept` either right now

Reply:
That's fine, it is just a suggestion.